### ISSUE TITLE
[3.0] Rewrite using React

### ISSUE THREAD

**Issue by milesj** [titon/toolkit#163]
Instead of recreating the wheel (yet again), we should port the entire framework to React. At the time of writing, there is no "go to" UI library for React yet, and Titon Toolkit hopes to fill that role. All of the current components will be ported to React, along with the inclusion of decorators, events, transitions, and even React functionality like prop types.
- [x] Accordion
- [x] Breadcrumb
- [ ] Button
- [ ] ButtonGroup
- [ ] Carousel
- [x] Checkbox (from InputCheckbox)
- [x] Divider
- [x] Drop
- [x] Flex
- [ ] Flyout
- [ ] Form
- [x] Grid
- [ ] Input
- [ ] InputGroup
- [ ] Label
- [x] LazyLoad
- [ ] Loader
- [x] Mask
- [ ] Matrix
- [ ] Modal
- [ ] Notice
- [x] OffCanvas
- [x] Pagination
- [ ] Pin
- [ ] Popover
- [x] Progress
- [x] Radio (from InputRadio)
- [x] Select (from InputSelect)
- [x] Step
- [x] Switch
- [x] Tab
- [ ] Table
- [ ] Toast
- [ ] Tooltip
- [ ] TypeAhead

**Comment by rshackleton** [comment]
Is the long term plan to only support the React version? We currently make use of Titon alongside Angular 1.5 and the migration to React would be incompatible.

**Comment by milesj** [comment]
@rshackleton Correct. The old 2x version is and will be support only, as no new development will occur there. The new 3x version will purely be React.
